client/interface.go

```diff
@@ -91,18 +91,25 @@ type ImageAPIClient interface {
 	BuildCachePrune(ctx context.Context, opts types.BuildCachePruneOptions) (*types.BuildCachePruneReport, error)
 	BuildCancel(ctx context.Context, id string) error
 	ImageCreate(ctx context.Context, parentReference string, options image.CreateOptions) (io.ReadCloser, error)
-	ImageHistory(ctx context.Context, image string) ([]image.HistoryResponseItem, error)
 	ImageImport(ctx context.Context, source image.ImportSource, ref string, options image.ImportOptions) (io.ReadCloser, error)
 	ImageInspectWithRaw(ctx context.Context, image string) (image.InspectResponse, []byte, error)
 	ImageList(ctx context.Context, options image.ListOptions) ([]image.Summary, error)
-	ImageLoad(ctx context.Context, input io.Reader, quiet bool) (image.LoadResponse, error)
 	ImagePull(ctx context.Context, ref string, options image.PullOptions) (io.ReadCloser, error)
 	ImagePush(ctx context.Context, ref string, options image.PushOptions) (io.ReadCloser, error)
 	ImageRemove(ctx context.Context, image string, options image.RemoveOptions) ([]image.DeleteResponse, error)
 	ImageSearch(ctx context.Context, term string, options registry.SearchOptions) ([]registry.SearchResult, error)
-	ImageSave(ctx context.Context, images []string) (io.ReadCloser, error)
 	ImageTag(ctx context.Context, image, ref string) error
 	ImagesPrune(ctx context.Context, pruneFilter filters.Args) (image.PruneReport, error)
+
+	// TODO: Change variadic opts into a single parameter
+	ImageHistory(ctx context.Context, image string, opts ...ImageHistoryOptions) ([]image.HistoryResponseItem, error)
+
+	// TODO: Change variadic opts into a single parameter
+	ImageSave(ctx context.Context, imageIDs []string, opts ...ImageSaveOptions) (io.ReadCloser, error)
+
+	// TODO: Replace ImageLoad with ImageLoadVNext
+	ImageLoad(ctx context.Context, input io.Reader, quiet bool) (image.LoadResponse, error)
+	ImageLoadVNext(ctx context.Context, input io.Reader, opts ImageLoadOptions) (image.LoadResponse, error)
```

Comment:
Any better way to do it? 🙈 

cc @thaJeztah @cpuguy83

Reply:
Should we put these in the same location as the other options? (`image.LoadOptions`), or won't that work?

Reply:
That would work, was considering it, but I think that (all) of them should really be separate types.

Sharing the same options structs between the client and the backend implementation in the daemon tends to leak the implementation details from the image service into the shared types.

Reply:
On the other hand.. I think it's actually the backend that does the bad thing here by reusing these. And Options struct in the `api/types/image` are actually expected to be consumed by the go client, and not the daemon's backend.

Reply:
Moved to `api/types/image`

Reply:
Ah, this will still break the interface matching for consumers 😭